### ISSUE TITLE
Error in bounded optimization using optimizer "trf" or "dogbox"

### ISSUE THREAD

**Issue by maxbergmann-cmd** [hyperspy/hyperspy#3240]
Hello everybody,

i'm trying to do a multifit of a EELS dataset using
m.multifit(optimizer="trf",bounded=True)

but it throws the error message:
ValueError: `bounds` must contain 2 elements.

Can somebody help me with that? Am i not supposed to use more than one free variable while optimizing with "trf"?

Best,
Max

**Comment by ericpre** [comment]
Can you provide the full trace error? It helps figuring out where it comes from?

**Comment by maxbergmann-cmd** [comment]
Sure:

```python
Traceback (most recent call last):
  File "/home/max/PycharmProjects/EELSFitting/venv/lib/python3.10/site-packages/IPython/core/interactiveshell.py", line 3508, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-1-ad9249c7449b>", line 1, in <module>
    prepare_bounds(bounds, x0.shape[0])
  File "/home/max/PycharmProjects/EELSFitting/venv/lib/python3.10/site-packages/scipy/optimize/_lsq/least_squares.py", line 96, in prepare_bounds
    lb, ub = [np.asarray(b, dtype=float) for b in bounds]
ValueError: too many values to unpack (expected 2)
Traceback (most recent call last):
  File "/home/max/Software/pycharm-community-2023.2.2/plugins/python-ce/helpers/pydev/pydevd.py", line 2199, in <module>
    main()
  File "/home/max/Software/pycharm-community-2023.2.2/plugins/python-ce/helpers/pydev/pydevd.py", line 2181, in main
    globals = debugger.run(setup['file'], None, None, is_module)
  File "/home/max/Software/pycharm-community-2023.2.2/plugins/python-ce/helpers/pydev/pydevd.py", line 1493, in run
    return self._exec(is_module, entry_point_fn, module_name, file, globals, locals)
  File "/home/max/Software/pycharm-community-2023.2.2/plugins/python-ce/helpers/pydev/pydevd.py", line 1500, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/home/max/Software/pycharm-community-2023.2.2/plugins/python-ce/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/home/max/PycharmProjects/EELSFitting/subtractCEdge.py", line 477, in <module>
    m.multifit(optimizer="trf",bounded=True)
  File "/home/max/PycharmProjects/EELSFitting/venv/lib/python3.10/site-packages/hyperspy/model.py", line 2043, in multifit
    self.fit(**kwargs)
  File "/home/max/PycharmProjects/EELSFitting/venv/lib/python3.10/site-packages/hyperspy/models/eelsmodel.py", line 341, in fit
    return Model1D.fit(self, **kwargs)
  File "/home/max/PycharmProjects/EELSFitting/venv/lib/python3.10/site-packages/hyperspy/model.py", line 1647, in fit
    self.fit_output = least_squares(
  File "/home/max/PycharmProjects/EELSFitting/venv/lib/python3.10/site-packages/scipy/optimize/_lsq/least_squares.py", line 805, in least_squares
    raise ValueError("`bounds` must contain 2 elements.")
ValueError: `bounds` must contain 2 elements.
python-BaseException
```

**Comment by ericpre** [comment]
Thank you @maxbergmann-cmd for the bug report, the issue is that the order of the bounds are incorrect for this optimizer `trf` but also `dogbox`, other optimizer works. https://github.com/hyperspy/hyperspy/issues/3240 fixes it and you will need to wait for the fix  to be included in a hyperspy release to use this optimizer with bounding. In the meantime, you can use other optimizer supporting bounds.

**Comment by ericpre** [comment]
Fixed in #3244.
